### PR TITLE
DOCSUP-10019: Support array in Parquet, ORC and Arrow formats

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #21770

### CODE REVIEW

File: docs/en/interfaces/formats.md [hunk 1]
```diff
@@ -1276,7 +1279,54 @@ To exchange data with Hadoop, you can use [HDFS table engine](../engines/table-e
 
 [Apache Arrow](https://arrow.apache.org/) comes with two built-in columnar storage formats. ClickHouse supports read and write operations for these formats.
 
-`Arrow` is Apache Arrow’s “file mode” format. It is designed for in-memory random access.
+`Arrow` is Apache Arrow’s "file mode" format. It is designed for in-memory random access.
+
+### Data Types Matching {#data_types-matching-4}
```
Comment (olgarev) [inline docs/en/interfaces/formats.md:1284]:
```suggestion
### Data Types Matching {#arrow_data_types-matching}
```

File: docs/en/interfaces/formats.md [hunk 2]
```diff
@@ -1276,7 +1279,54 @@ To exchange data with Hadoop, you can use [HDFS table engine](../engines/table-e
...
+
+Arrays can be nested and can have a value of the `Nullable` type as an argument.
+
+ClickHouse supports configurable precision of the `Decimal` type. The `INSERT` query treats the Arrow `DECIMAL` type as the ClickHouse `Decimal128` type.
+
+Unsupported Arrow data types: `TIME32`, `FIXED_SIZE_BINARY`, `JSON`, `UUID`, `ENUM`.
+
+The data types of ClickHouse table columns do not have to match the corresponding Arrow data fields. When inserting data, ClickHouse interprets data types according to the table above and then [casts](../sql-reference/functions/type-conversion-functions.md#type_conversion_function-cast) the data to the data type set for the ClickHouse table column.
+
+### Inserting Data {#inserting-data-3}
```
Comment (olgarev) [inline docs/en/interfaces/formats.md:1315]:
```suggestion
### Inserting Data {#arrow-inserting-data}
```

File: docs/en/interfaces/formats.md [hunk 3]
```diff
@@ -1276,7 +1279,54 @@ To exchange data with Hadoop, you can use [HDFS table engine](../engines/table-e
...
+
+You can insert Arrow data from a file into ClickHouse table by the following command:
+
+``` bash
+$ cat filename.arrow | clickhouse-client --query="INSERT INTO some_table FORMAT Arrow"
+```
+
+### Selecting Data {#selecting-data-3}
```
Comment (olgarev) [inline docs/en/interfaces/formats.md:1323]:
```suggestion
### Selecting Data {#selecting-data-arrow}
```

File: docs/ru/interfaces/formats.md [hunk 4]
```diff
@@ -1168,10 +1168,13 @@ SELECT * FROM topic1_stream;
...
 | —                             | [FixedString](../sql-reference/data-types/fixedstring.md) | `STRING`                      |
 | `DECIMAL`                     | [Decimal](../sql-reference/data-types/decimal.md)         | `DECIMAL`                     |
+| `LIST`                        | [Array](../sql-reference/data-types/array.md)             | `LIST`                        |
+
+Массивы могут быть вложенными и иметь в качестве аргумента значение типа `Nullable`.
 
 ClickHouse поддерживает настраиваемую точность для формата `Decimal`. При обработке запроса `INSERT`, ClickHouse обрабатывает тип данных Parquet `DECIMAL` как `Decimal128`.
 
-Неподдержанные типы данных Parquet: `DATE32`, `TIME32`, `FIXED_SIZE_BINARY`, `JSON`, `UUID`, `ENUM`.
+Неподдержанные типы данных Parquet: `TIME32`, `FIXED_SIZE_BINARY`, `JSON`, `UUID`, `ENUM`.
```
Comment (olgarev) [inline docs/ru/interfaces/formats.md:1177]:
```suggestion
Неподдерживаемые типы данных Parquet: `TIME32`, `FIXED_SIZE_BINARY`, `JSON`, `UUID`, `ENUM`.
```

File: docs/ru/interfaces/formats.md [hunk 5]
```diff
@@ -1197,6 +1200,53 @@ $ clickhouse-client --query="SELECT * FROM {some_table} FORMAT Parquet" > {some_
 
 `Arrow` — это Apache Arrow's "file mode" формат. Он предназначен для произвольного доступа в памяти.
 
+### Соответствие типов данных {#sootvetstvie-tipov-dannykh-4}
```
Comment (olgarev) [inline docs/ru/interfaces/formats.md:1203]:
```suggestion
### Соответствие типов данных {#data_types-matching-arrow}
```

File: docs/ru/interfaces/formats.md [hunk 6]
```diff
@@ -1197,6 +1200,53 @@ $ clickhouse-client --query="SELECT * FROM {some_table} FORMAT Parquet" > {some_
...
+
+Массивы могут быть вложенными и иметь в качестве аргумента значение типа `Nullable`.
+
+ClickHouse поддерживает настраиваемую точность для формата `Decimal`. При обработке запроса `INSERT`, ClickHouse обрабатывает тип данных Arrow `DECIMAL` как `Decimal128`.
+
+Неподдержанные типы данных Arrow: `TIME32`, `FIXED_SIZE_BINARY`, `JSON`, `UUID`, `ENUM`.
+
+Типы данных столбцов в ClickHouse могут отличаться от типов данных соответствующих полей файла в формате Arrow. При вставке данных, ClickHouse интерпретирует типы данных в соответствии с таблицей выше, а затем [приводит](../sql-reference/functions/type-conversion-functions/#type_conversion_function-cast) данные к тому типу, который установлен для столбца таблицы.
+
+### Вставка данных {#vstavka-dannykh-3}
```
Comment (olgarev) [inline docs/ru/interfaces/formats.md:1234]:
```suggestion
### Вставка данных {#inserting-data-arrow}
```

File: docs/ru/interfaces/formats.md [hunk 7]
```diff
@@ -1197,6 +1200,53 @@ $ clickhouse-client --query="SELECT * FROM {some_table} FORMAT Parquet" > {some_
...
+
+Чтобы вставить в ClickHouse данные из файла в формате Arrow, используйте команду следующего вида:
+
+``` bash
+$ cat filename.arrow | clickhouse-client --query="INSERT INTO some_table FORMAT Arrow"
+```
+
+### Вывод данных {#vyvod-dannykh-3}
```
Comment (olgarev) [inline docs/ru/interfaces/formats.md:1242]:
```suggestion
### Вывод данных {#selecting-data-arrow}
```

File: docs/ru/interfaces/formats.md [hunk 8]
```diff
@@ -1197,6 +1200,53 @@ $ clickhouse-client --query="SELECT * FROM {some_table} FORMAT Parquet" > {some_
...
+| `DATE32`                    | [Date](../sql-reference/data-types/date.md)         | `UINT16`                    |
+| `DATE64`, `TIMESTAMP`       | [DateTime](../sql-reference/data-types/datetime.md) | `UINT32`                    |
+| `STRING`, `BINARY`          | [String](../sql-reference/data-types/string.md)     | `UTF8`                      |
+| `STRING`, `BINARY`          | [FixedString](../sql-reference/data-types/fixedstring.md)   | `UTF8`                        |
+| `DECIMAL`                   | [Decimal](../sql-reference/data-types/decimal.md)   | `DECIMAL`                   |
+| `LIST`                      | [Array](../sql-reference/data-types/array.md)       | `LIST`                      |
+
+Массивы могут быть вложенными и иметь в качестве аргумента значение типа `Nullable`.
+
+ClickHouse поддерживает настраиваемую точность для формата `Decimal`. При обработке запроса `INSERT`, ClickHouse обрабатывает тип данных Arrow `DECIMAL` как `Decimal128`.
```
Comment (olgarev) [inline docs/ru/interfaces/formats.md:1228]:
```suggestion
ClickHouse поддерживает настраиваемую точность для формата `Decimal`. При выполнении запроса `INSERT` ClickHouse обрабатывает тип данных Arrow `DECIMAL` как `Decimal128`.
```

File: docs/ru/interfaces/formats.md [hunk 9]
```diff
@@ -1197,6 +1200,53 @@ $ clickhouse-client --query="SELECT * FROM {some_table} FORMAT Parquet" > {some_
...
+
+Неподдержанные типы данных Arrow: `TIME32`, `FIXED_SIZE_BINARY`, `JSON`, `UUID`, `ENUM`.
```
Comment (olgarev) [inline docs/ru/interfaces/formats.md:1230]:
```suggestion
Неподдерживаемые типы данных Arrow: `TIME32`, `FIXED_SIZE_BINARY`, `JSON`, `UUID`, `ENUM`.
```

File: docs/ru/interfaces/formats.md [hunk 10]
```diff
@@ -1197,6 +1200,53 @@ $ clickhouse-client --query="SELECT * FROM {some_table} FORMAT Parquet" > {some_
...
+
+Типы данных столбцов в ClickHouse могут отличаться от типов данных соответствующих полей файла в формате Arrow. При вставке данных, ClickHouse интерпретирует типы данных в соответствии с таблицей выше, а затем [приводит](../sql-reference/functions/type-conversion-functions/#type_conversion_function-cast) данные к тому типу, который установлен для столбца таблицы.
```
Comment (olgarev) [inline docs/ru/interfaces/formats.md:1232]:
```suggestion
Типы данных столбцов в ClickHouse могут отличаться от типов данных соответствующих полей файла в формате Arrow. При вставке данных ClickHouse интерпретирует типы данных в соответствии с таблицей выше, а затем [приводит](../sql-reference/functions/type-conversion-functions/#type_conversion_function-cast) данные к тому типу, который установлен для столбца таблицы.
```

**Comment by sevirov** [comment]
@Avogar Please, review this PR.

**Review by Avogar — APPROVED** [review]

**Pull request merged**